[lexical-selection] Bug Fix: Correct caret movement in vertical-rl writing mode

kirandash opened the pull request:
## Description

### Current behavior:
In vertical-rl writing mode, when navigating between empty paragraphs, arrow keys move in the opposite direction from expected and also blocks the navigation to previous nodes. Right arrow moves left and left arrow moves right and is blocked. This only happens in empty paragraphs (`<p><br></p>`).

### Changes in this PR:
This PR fixes the incorrect caret movement in vertical-rl writing mode by:
1. Modifying `$moveCharacter` to correctly handle vertical writing mode orientation
2. Adding support for detecting vertical writing mode in the editor
3. Ensuring correct behavior for empty paragraphs by inverting arrow key directions
4. Passing the editor instance to arrow key handlers in both rich-text and plain-text plugins

Closes #7363

## Test plan

### Before

Arrow keys behave incorrectly in vertical-rl writing mode:
- Right arrow key moves left (instead of right), keep pressing left but arrow does not navigate to previous paragraphs
- Left arrow key moves right (instead of left), keep pressing right but arrow does not navigate to previous paragraphs
This incorrect behavior only occurs in empty paragraphs.

https://github.com/user-attachments/assets/31a009a4-1968-453c-a72c-ce6f0a336fa1

### After

Arrow keys now behave correctly in vertical-rl writing mode:
- Right arrow key moves right
- Left arrow key moves left

https://github.com/user-attachments/assets/9644d0f8-3aab-4bee-b9d5-8b9437c71f0d

### Testing Note
I attempted to add E2E tests to verify this fix, but encountered challenges in getting them to pass. This is likely due to the complexity of simulating and asserting cursor movement behavior in vertical writing mode.

Instead, I've included a video demonstration showing the working fix in the UI.

etrepum reviewed (commented):
Let's refactor it to not pass the editor explicitly, you can use `$getEditor()` in the one place it's needed.

Similarly I think that the vertical writing mode should be based on where the selection is (similar to how $isParentElementRTL works).

Probably this vertical writing mode should be part of the lexical model rather than only in DOM but it would be a reasonable fix to do it this way in the meantime

kirandash commented:
> Let's refactor it to not pass the editor explicitly, you can use `$getEditor()` in the one place it's needed.
> 
> Similarly I think that the vertical writing mode should be based on where the selection is (similar to how $isParentElementRTL works).
> 
> Probably this vertical writing mode should be part of the lexical model rather than only in DOM but it would be a reasonable fix to do it this way in the meantime

Thanks for the review! I've pushed a commit [f9dbfbd](https://github.com/facebook/lexical/pull/7439/commits/f9dbfbd156b7e1d8fe2a05060826ae4cd8416ed7) that implements your suggestions:

1. Refactored to remove the explicit `editor` parameter and use `$getEditor()` internally
2. Updated the vertical writing mode check to be selection-based, similar to `$isParentElementRTL`

Please let me know if you'd like me to make any other changes to this PR!

etrepum commented on packages/lexical-selection/src/range-selection.ts:
```diff
@@ -416,6 +417,27 @@ export function $wrapNodesImpl(
+  if (domElement === null) {
+    return false;
+  }
+  const computedStyle = window.getComputedStyle(domElement);
```
When iframes are in use, the editor's window may not match the global window

```suggestion
  const computedStyle = getWindow(editor).getComputedStyle(domElement);
```

Could also use `domElement.ownerDocument.defaultView` instead of `getWindow(editor)`, should be the same thing at least under the current constraints of how lexical DOM is allowed to work

kirandash replied:
Thanks for the suggestion! I've pushed a [commit](https://github.com/facebook/lexical/pull/7439/commits/147dcf3fe3377b75e0b9f9b151398a3daf63da5f) using `domElement.ownerDocument.defaultView`.

etrepum reviewed (commented):
Haven't done any manual testing yet but there's one small issue with the use of window

etrepum reviewed (approved):
I suspect that this will be revisited but it does seem like an improvement. I don't have any experience with how the vertical-rl writing mode is supposed to work
